tools/bus_space/bus.c

```diff
@@ -227,10 +228,11 @@ bs_unmap(int rid)
 }
 
 int
-bs_write(int rid, off_t ofs, void *buf, ssize_t bufsz)
+bs_write(int rid, off_t ofs, void *buf64, ssize_t bufsz)
```

Comment:
same comments all the way through for bs_write as bs_read.
To support 64 bit types you just need a case 8: in both of these functions.